oeps/oep-0028-content-theming-in-react.rst

```diff
@@ -0,0 +1,246 @@
+| Status          | Draft                                                          |
++-----------------+----------------------------------------------------------------+
+| Type            | Architecture                                                   |
++-----------------+----------------------------------------------------------------+
+| Created         | 2018-08-17                                                     |
++-----------------+----------------------------------------------------------------+
+| `Review Period` | 2018-08-17 - 2018-09-02                                        |
++-----------------+----------------------------------------------------------------+
+
+Abstract
+-------
+
+OEP-0023 [1] addresses the styling part of a theming system based on React. This proposal adds guidelines for content modification aspect to the React based theming system
+
+Context
+-------
+
+Open edX instances frequently wish to customize not only the colors and styles of the Open edX UI, but also to add, remove, and modify significant parts of the UI to suit their use cases and branding. Doing this in a way that is understandable, clean, and easy to maintain can be a challenge unless the original UI is designed with this in mind.
+
+Another pain of theming system is providing access to unrelated data to any component. This data may not be used by earlier components but is required in a particular theme. It becomes an important concern for writing a theming system where a global access to data to all the components in the UI system should be provided.
+
+Decision
+--------
+
+* We will build small modular components which will be non-customizable, stateless and contain HTML, and certain logic. We can see an example of a smaller component SiteLogo, which will later be used in a customizable component _Header.
+
+.. code-block:: js
+
+    const SiteLogo = (props) => {
+        return (
+            <img src={props.url} alt={props.altText} />
+        );
+    }
+
+* We will build the complete UI out of the above small modular components. We will generally use composition to build a customizable component. However, inheritance can be used at times. These customizable components will neither contain HTML nor logic. We can see an example of a customizable component called the _Header component, which is formed by the composition of SiteLogo, MainNav and UserAvatar component.
+
+.. code-block:: js
+
+    class _Header extends React.PureComponent {
+        render() {
+            return (
+                <header>
+                    <SiteLogo/>
+                    <MainNav/>
+                    <UserAvatar/>
+                </header>
+            );
+        }
+    }
+
+* We will follow a 2 step approach to create any UI Component. We will first break down a complex component into smaller non-customizable components. We will then compose these smaller components to form a complex UI.
```

Comment:
I think this can be rephrased a little bit. Currently, it sounds like you already have a component, which is why you are breaking it down. 

Perhaps:
When creating any complex UI Component we will try to break it down into the smallest usable components, and build the component out of these small reusable pieces.

Reply:
I have addressed this.